packages/unpublished-adapter-encapsulation-test-app/tests/integration/queries-test.js

```diff
@@ -0,0 +1,222 @@
+    // if IDENTIFIERS is turned on
+    delete expectedResult.data.lid;
+
+    assert.equal(findRecordCalled, 1, 'findRecord is called once');
+    assert.deepEqual(record.serialize(), expectedResult, 'findRecord returns expected result');
+  });
+
+  test('store.findAll calls adapter.findAll w/correct args', async function(assert) {
+    let findAllCalled = 0;
+    let expectedResult = {
+      data: [
+        {
+          id: '12',
+          type: 'person',
+          attributes: {
+            firstName: 'Gaurav',
+            lastName: 'Munjal',
+          },
+        },
+        {
+          id: '19',
+          type: 'person',
+          attributes: {
+            firstName: 'Chris',
+            lastName: 'Thoburn',
+          },
+        },
+      ],
+    };
+    let { owner } = this;
+    let store = owner.lookup('service:store');
+
+    class TestFindAllAdapter extends EmberObject {
+      findAll(passedStore, type, sinceToken, snapshot) {
+        findAllCalled++;
+
+        assert.equal(passedStore, store, 'instance of store is passed to findAll');
+        assert.equal(type, Person, 'model is passed to findAll');
+        assert.strictEqual(sinceToken, null, 'sinceToken passed to findAll is null');
+        assert.equal(snapshot.modelName, 'person', 'snapshot is passed to findAll with correct modelName');
+        assert.equal(snapshot.length, 0, 'snapshot is passed to findAll represnts empty array');
+
+        return resolve(expectedResult);
+      }
+    }
+
+    owner.register('adapter:application', TestFindAllAdapter);
+
+    let manyArray = await store.findAll('person');
+
+    // This code is a workaround for issue https://github.com/emberjs/data/issues/6758
+    // expectedResult is unexpectedly mutated during store.findAll
+    // if IDENTIFIERS is turned on
+    expectedResult.data.forEach(result => {
+      delete result.lid;
+    });
+
+    let result = manyArray.toArray().map(person => person.serialize());
+    expectedResult = expectedResult.data.map(person => ({ data: person }));
+
+    assert.equal(findAllCalled, 1, 'findAll is called once');
+    assert.deepEqual(result, expectedResult, 'findAll returns expected result');
+  });
+
+  test('store.queryRecord calls adapter.queryRecord w/correct args', async function(assert) {
+    let queryRecordCalled = 0;
+    let expectedResult = {
+      data: {
+        id: '12',
+        type: 'person',
+        attributes: {
+          firstName: 'Gaurav',
+          lastName: 'Munjal',
+        },
+      },
+    };
+    let { owner } = this;
+    let store = owner.lookup('service:store');
+
+    class TestQueryRecordAdapter extends EmberObject {
+      queryRecord(passedStore, type, query) {
+        queryRecordCalled++;
+
+        assert.equal(passedStore, store, 'instance of store is passed to queryRecord');
+        assert.equal(type, Person, 'model is passed to queryRecord');
+        assert.deepEqual(query, { firstName: 'Gaurav' }, 'query is passed to queryRecord');
```

Comment:
While not implemented by the default adapters, the full signature for `queryRecord` is 4 args

```ts
  queryRecord(store: Store, schema: ModelClass, query, options): Promise<unknown>;
```